pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -567,11 +566,21 @@ func (c *PostableUserConfig) UnmarshalYAML(value *yaml.Node) error {
 	return nil
 }
 
+type Provenance string
+
+const (
+	// ProvenanceNone reflects the provenance when no provenance is stored
+	// for the requested object in the database.
+	ProvenanceNone Provenance = ""
+	ProvenanceAPI  Provenance = "api"
+	ProvenanceFile Provenance = "file"
```

Comment:
Part of the principle we had was to keep the API layer's understanding of `provenance` more like a string than an enum. There was an original idea of supporting more provenance types, or allowing custom provenance types...

WDYT about doing something like `type Provenance string` alone in the API layer, and not specifying any actual constant values?

Most of the code in this layer cares about the type itself more than it cares about the values - and the conversion func can handle the matching.

Reply:
+1 for that

Reply:
https://github.com/grafana/grafana/pull/63594/commits/c6cb7d297e69a6ce4c3905b8711b3134d65964d2